assignment3.py

```diff
@@ -0,0 +1,73 @@
+class Dictionary(object):
+	def __init__(self, dictionary=None):
+		self.dictionary = []
+		# self.prefixes = []
+
+	def insert_words(self, word_list):
+		self.dictionary = word_list
+
+	def is_word(self, string):
+		return (string in self.dictionary)
+
+	def is_prefix(self, string):
+		for word in self.dictionary:
+			if len(word) >= len(string) and word[0:len(string)] == string:
+				return True
+		return False
+
+
+result_set = []
+def solution(n_rows, n_columns, matrix, dict_object):
+	word_sofar = ''
+	for i in range(n_rows):
+		for j in range(n_columns):
+			visited = []
+			find_word_for_cell(n_rows, n_columns, matrix, dict_object, i, j, word_sofar, visited)
+
+	print(result_set)
+
+
+def find_word_for_cell(n_rows, n_columns, matrix, dict, row_number, col_number, word_sofar, visited ):\
+
+	word_sofar = word_sofar + matrix[row_number][col_number]
+
+	if(dict.is_word(word_sofar)):
+		# print(word_sofar)
+		result_set.append(word_sofar)
+
+	if(not dict.is_prefix(word_sofar)):
+		return
+
+	visited.append([row_number,col_number])
+
+	# left up
```

Comment:
While this block of ifs is correct, can you maybe think of a more maintainable way to implement it? (Generalize all the 8 cases)